pkg/ccl/backupccl/restore_job.go

```diff
@@ -290,11 +289,9 @@ func restore(
 		return emptyRowCount, err
 	}
 
-	on231 := clusterversion.V23_1.Version().LessEq(job.Payload().CreationClusterVersion)
```

Comment:
Is this actually safe to remove? direct upgrades from 23.1 are still supported

Even if they weren't, we don't have any promises a job created in an older version won't be resumed by a newer version since we don't wait for jobs during upgrades.